Substra/substra-frontend
# feat: add SSO

## Description

Add mechanism to handle a new authentication flow: single sign-on (SSO) through OpenID Connect (OIDC).

Changes:
 - on the login page, we query the `/info` endpoint to see whether SSO is available, and if so present a button. All the button does is redirect to the backend at `/oidc/authenticate?next={some frontend page}`. The backend does everything else
 - add the current username in the header menu, because with SSO the username is not obvious
 - add an "API tokens" modal (in the header menu) so SSO users can use the library (they don't have a password).
   
   In the future we will deprecate logging in on the command line, and that modal will be a nice interface for creating and managing API tokens. Probably.

Issues: _(to be fixed in a future PR, presumably)_
 - active tokens make too many requests and should go through Redux
 - API tokens modal is ugly



## How to test

Only the actual login requires special set-up.

Set up an SSO-capable backend: https://github.com/Substra/substra-backend/pull/609

Then just edit `vite.config.ts` to point to `http://substra-backend.org-1.com` (without the port number, because we're hosting on the cluster rather than in isolated mode). That should be it.

It's also possible to test the frontend in isolated mode but that requires more set-up on the backend so I'll let you figure it out.

## To do

- [x] ~~investigate login/logout redirects (using the logout button and then logging in again works weird)~~ can't reproduce
